### PR TITLE
test: team user services

### LINKED ISSUE #1320: [TEST]: team user service
## Description

Add tests and refactor the team user services.

### PR DESCRIPTION
**Pull request by patricia-mdias**
Relates to:
- #1320 

## Proposed Changes

  - Add tests to team user services and use cases.

This pull request closes #1320

### CODE REVIEW

File: backend/src/modules/teamUsers/services/create.team.user.service.ts [hunk 1]
```diff
@@ -13,26 +13,23 @@ export default class CreateTeamUserService implements CreateTeamUserServiceInter
...
+	): Promise<TeamUser[]> {
 		const teamUsersToSave = teamId
 			? teamUsers.map((teamUser) => ({ ...teamUser, team: teamId }))
 			: teamUsers;
 
-		const teamUsersSaved = await this.teamUserRepository.insertMany(teamUsersToSave);
+		const teamUsersSaved = await this.teamUserRepository.insertMany(teamUsersToSave, withSession);
 
-		if (teamUsersSaved.length < 1) throw new BadRequestException(INSERT_FAILED);
+		if (teamUsersSaved.length < teamUsers.length) throw new BadRequestException(INSERT_FAILED);
```
Comment (CatiaBarroco-xgeeks) [inline backend/src/modules/teamUsers/services/create.team.user.service.ts:27]: wouldn't it be better to put it inside a constant (like: `cont isNameee = teamUsersSaved.length < teamUsers.length`) with what it's supposed to validate?
Maybe it gets a better reading, what do you think?

File: backend/src/modules/teamUsers/applications/create-team-user.use-case.spec.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { createTeamUserUseCase } from './../teamusers.providers';
+import { Test, TestingModule } from '@nestjs/testing';
+import { DeepMocked, createMock } from '@golevelup/ts-jest';
+import { UseCase } from 'src/libs/interfaces/use-case.interface';
+import * as TeamUsers from 'src/modules/teamUsers/interfaces/types';
```
Comment (CatiaBarroco-xgeeks) [inline backend/src/modules/teamUsers/applications/create-team-user.use-case.spec.ts:5]: Is it necessary to import everything and not just what is needed for this component?
At the moment there is only one constant inside this folder, but in terms of project scalability, if we put something inside this file we will be importing everything and we only need a specific part of the "types" file
What do you think about it?

File: backend/src/modules/teamUsers/services/create.team.user.service.ts (hunk 1 above)
Comment (patricia-mdias) [reply backend/src/modules/teamUsers/services/create.team.user.service.ts:27]: That's a great idea! Thank you for the input.

File: backend/src/modules/teamUsers/applications/create-team-users.use-case.ts [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
 @Injectable()
 export class CreateTeamUsersUseCase implements UseCase<TeamUserDto[], TeamUser[]> {
 	constructor(
-		@Inject(TYPES.repositories.TeamUserRepository)
-		private readonly teamUserRepository: TeamUserRepositoryInterface
+		@Inject(TYPES.services.CreateTeamUserService)
+		private createTeamUserService: CreateTeamUserServiceInterface
 	) {}
 
 	async execute(teamUsers: TeamUserDto[]): Promise<TeamUser[]> {
```
Comment (nunocaseiro) [inline backend/src/modules/teamUsers/applications/create-team-users.use-case.ts:15]:
```suggestion
	execute(teamUsers: TeamUserDto[]): Promise<TeamUser[]> {
```

File: backend/src/modules/teams/applications/create.team.application.ts [hunk 4]
```diff
@@ -1,25 +1,16 @@
...
-import * as TeamUsers from 'src/modules/teamUsers/interfaces/types';
 
 @Injectable()
 export class CreateTeamApplication implements CreateTeamApplicationInterface {
 	constructor(
 		@Inject(TYPES.services.CreateTeamService)
-		private createTeamService: CreateTeamServiceInterface,
-		@Inject(TeamUsers.TYPES.services.CreateTeamUserService)
-		private createTeamUserService: CreateTeamUserServiceInterface
+		private createTeamService: CreateTeamServiceInterface
```
Comment (nunocaseiro) [inline backend/src/modules/teams/applications/create.team.application.ts:11]:
```suggestion
		private readonly createTeamService: CreateTeamServiceInterface
```

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

File: backend/src/modules/teamUsers/services/create.team.user.service.ts (hunk 1 above)
Comment (CatiaBarroco-xgeeks) [reply backend/src/modules/teamUsers/services/create.team.user.service.ts]: Done https://github.com/xgeekshq/split/pull/1353/commits/c6129815280dc801ec856ad928da95a88b800a6c

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀

**Pull request merged**